### PR TITLE
Add heap and stack to test

### PR DESCRIPTION
**Pull request by danghai**
PR to add test_heap and test_stack to tests.

### CODE REVIEW

File: heap/binary_heap.py [hunk 1]
```diff
@@ -51,7 +50,7 @@ def min_child(self,i):
     @abstractmethod
     def remove_min(self,i):
         pass
-class Binary_Heap(AbstractHeap):
+class binary_heap(AbstractHeap):
```
Comment (goswami-rahul) [inline heap/binary_heap.py]: This should be renamed to `BinaryHeap`.

File: stack/stack.py [hunk 2]
```diff
@@ -49,7 +46,7 @@ def peek(self):
         pass
 
 
-class ArrayStack(AbstractStack):
+class Array_stack(AbstractStack):
```
Comment (goswami-rahul) [inline stack/stack.py]: I think it was correct before? `ArrayStack`.

File: stack/stack.py [hunk 3]
```diff
@@ -101,7 +98,7 @@ def __init__(self, value):
         self.next = None
 
 
-class LinkedListStack(AbstractStack):
+class Linkedlist_stack(AbstractStack):
```
Comment (goswami-rahul) [inline stack/stack.py]: `LinkedListStack`

File: tests/test_heap.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+from heap.binary_heap import binary_heap
+from heap.skyline import get_skyline
+from heap.sliding_window_max import max_sliding_window
+
+import unittest
+
+class Test_binary_heap(unittest.TestCase):
```
Comment (goswami-rahul) [inline tests/test_heap.py]: rename to `TestBinaryHeap`

File: tests/test_stack.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        self.assertFalse(is_valid("[[[]]"))
+        self.assertTrue(is_valid("{([])}"))
+        self.assertFalse(is_valid("(}"))
+
+    def test_simplify_path(self):
+        p = '/my/name/is/..//keon'
+        self.assertEqual('/my/name/keon', simplify_path(p))
+
+
+class Test_Stack(unittest.TestCase):
```
Comment (goswami-rahul) [inline tests/test_stack.py]: Why another class for it?

Comment (danghai) [reply tests/test_stack.py]: Because I think stack.py is a class with many methods. Other files is just function. I think it is more clear :)

Comment (christianbender) [reply tests/test_stack.py]: @danghai Please use ```TitleCase``` for class names.

**Comment by danghai** [comment]
I have updated it!

**Pull request merged**